cmd/gpu-kubelet-plugin/driver.go

```diff
@@ -93,6 +97,16 @@ func NewDriver(ctx context.Context, config *Config) (*driver, error) {
 	}
 	driver.healthcheck = healthcheck
 
+	if featuregates.Enabled(featuregates.DeviceHealthCheck) {
+		deviceHealthMonitor, err := newDeviceHealthMonitor(ctx, config, state.allocatable, state.nvdevlib)
+		if err != nil {
+			return nil, fmt.Errorf("start deviceHealthMonitor: %w", err)
+		}
+
+		driver.deviceHealthMonitor = deviceHealthMonitor
+		go driver.deviceHealthEvents(ctx, config.flags.nodeName)
```

Comment:
Something needs to wait somehwere / somehow for this go routine to exit. The typical pattern is with a waitgroup.